### ISSUE TITLE
Some Clarification needed

### ISSUE THREAD

**Issue by jaideep11061982** [chenghuige/U.S.-Patent-Phrase-to-Phrase-Matching#1]
Dear Cheng
Where you are covering the below thing of your solution in your notebooks. 
Many thanks for sharing.
Congrats for your winning streak.

CV strategy
Groupby anchor and stratify by score, also there are some words occur in both anchor and target, make sure to put them in the same fold.

NN model detail
a. Pearson loss worked best for me
b. 5 epochs training, start AWP training from the 2nd epoch.
AWP helps a lot in all my nlp contests recently.
c. Groupby['anchor', 'context'] ['target'] -> targets, add to input(anchor[SEP]target[SEP]CPC_TEXT[SEP]targets) produce best model
Groupby['anchor', 'context[0]'] ['target'] -> targets, add to input help ensemble a lot, let me define context[0] as sector, it is like F21 -> F
Remember to exclude current target from targets.

f. Using different learning rates for bert (2e-5, 3e-5) and other parts(1e-3), especially useful when adding LSTM which need large lr.
g. Add BI-LSTM header help a lot.
Deberta-v3-large CV 858-> 861 “prompt is all you need" gave me a hint we do not need to finetune/change bert model a lot, so I tried to add LSTM on top of bert and freeze bert embedding layer.

**Comment by chenghuige** [comment]
@jaideep11061982  I just notice you posed this issue.
Most from src/torch/preprocess.py for example cv strategy,Groupby['anchor', 'context'] ['target'] -> targets
Model file src/torch/model.py
Pearson loss usage in src/torch/loss.py 
Pearson loss code in pikachu/utils/lele/losses/losses.py
AWP code in pikachu/utils/lele/util.py
